src/Stratis.Bitcoin.Features.Consensus/Rules/CommonRules/CalculateStakeRule.cs

```diff
@@ -30,6 +30,15 @@ public override Task RunAsync(RuleContext context)
 
             context.NextWorkRequired = this.PosParent.StakeValidator.GetNextTargetRequired(this.PosParent.StakeChain, context.ValidationContext.ChainedHeader.Previous, context.Consensus, posRuleContext.BlockStake.IsProofOfStake());
```

Comment:
Maybe we no longer need NextWorkRequired  in the context after this change

Reply:
But wait a second, This is partial validation rule and we would like the difficulty to be checked in header validator

Reply:
Hmm this requires stake calculation, yeah I remember we said its a header rule.

Reply:
ok done